### PR TITLE
TOC selection scrolling offset fixes

### PR DESCRIPTION
**Pull request by montehurd**
https://phabricator.wikimedia.org/T209442

Fixes for TOC selection scrolling offsets: 
- iOS 12 when selecting non-root sections
- iOS 11 and 12 when nav bar hidden

### CODE REVIEW

File: WMF Framework/NavigationBar.swift [hunk 1]
```diff
@@ -432,7 +432,11 @@ public class NavigationBar: SetupView, FakeProgressReceiving, FakeProgressDelega
     var extendedViewHeight: CGFloat {
         return extendedView.frame.size.height
     }
-    
+
+    @objc var backgroundViewY: CGFloat {
```
Comment (joewalsh) [inline WMF Framework/NavigationBar.swift]: Could this be named `hiddenHeight` for consistency with `visibleHeight`? `backgroundView` is an implementation detail and private, so having this named `backgroundViewY` could be confusing. `hiddenHeight` could be set the same way as `visibleHeight` in `layoutSubviews`, equal to `barTransformHeight + extendedViewTransformHeight + underBarTransformHeight`. `totalTransform` is then `0 - hiddenHeight` and `self.navigationBar.hiddenHeight` is added to the calculation in `WebViewController`

Comment (montehurd) [reply WMF Framework/NavigationBar.swift]: Sounds good 👍 Will update.

Comment (joewalsh) [reply WMF Framework/NavigationBar.swift]: it should be positive - `hiddenHeight` == the total height that is hidden, similar to how `visibleHeight` == the total height that is visible.  It should be `barTransformHeight + extendedViewTransformHeight + underBarTransformHeight` and the transform is negative, `0 - hiddenHeight`

Comment (montehurd) [reply WMF Framework/NavigationBar.swift]: Oops I think I should have had coffee before reading your initial comment ;) Will remedy that now hehe.

Comment (montehurd) [reply WMF Framework/NavigationBar.swift]: Fixed!

File: Wikipedia/Code/WebViewController.m [hunk 2]
```diff
@@ -789,7 +789,7 @@ - (void)scrollToFragment:(NSString *)fragment animated:(BOOL)animated {
         [self.webView getScrollViewRectForHtmlElementWithId:fragment
                                                  completion:^(CGRect rect) {
                                                      if (!CGRectIsNull(rect)) {
-                                                         [self.webView.scrollView wmf_safeSetContentOffset:CGPointMake(self.webView.scrollView.contentOffset.x, rect.origin.y)
+                                                         [self.webView.scrollView wmf_safeSetContentOffset:CGPointMake(self.webView.scrollView.contentOffset.x, rect.origin.y + [self iOS12yOffsetHack] - (0 - self.delegate.navigationBar.hiddenHeight))
```
Comment (joewalsh) [inline Wikipedia/Code/WebViewController.m]: `1 - (0 - x)` is the same as `1 + x`

Comment (montehurd) [reply Wikipedia/Code/WebViewController.m]: lol oops

Comment (montehurd) [reply Wikipedia/Code/WebViewController.m]: just a sec..

**Pull request merged**